### PR TITLE
mgr/dashboard: adds links to the services 

### PR DESCRIPTION
**Pull request by Pegonzal**
[screen-capture (3).webm](https://github.com/ceph/ceph/assets/106810838/03799695-31f4-4f49-b7e2-7c2111e7b781)

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by cloudbehl** [comment]
Lets add the link for normal usecase as well. If we don't have ngnix deployed we can pick it up from config and show it to user.

File: src/pybind/mgr/dashboard/controllers/prometheus.py [hunk 1]
```diff
@@ -135,6 +135,14 @@ def list(self, cluster_filter=False, **params):
...
         return self.alert_proxy('GET', '/alerts', params)
 
+    @RESTController.Collection(method='GET')
+    def url(self):
+        return Settings.PROMETHEUS_API_HOST
+
+    @RESTController.Collection(method='GET', path='/alertmanager_url')
+    def alertmanager_url(self):
+        return Settings.ALERTMANAGER_API_HOST
+
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/controllers/prometheus.py:145]: we don't need this right? there are existing api's that already returns the url if I am not wrong?

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/services.component.ts [hunk 2]
```diff
@@ -71,6 +74,8 @@ export class ServicesComponent extends ListWithDetails implements OnChanges, OnI
   isLoadingServices = false;
   selection: CdTableSelection = new CdTableSelection();
   icons = Icons;
+  service_urls = { grafana: '', prometheus: '', alertmanager: '' };
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/services.component.ts:77]:
```suggestion
  serviceUrls = { grafana: '', prometheus: '', alertmanager: '' };
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/services.component.ts [hunk 3]
```diff
@@ -178,6 +185,10 @@ export class ServicesComponent extends ListWithDetails implements OnChanges, OnI
       this.orchStatus = status;
       this.showDocPanel = !status.available;
     });
+
+    this.configureServiceUrl('api/grafana/url', 'grafana');
+    this.configureServiceUrl('api/prometheus/url', 'prometheus');
+    this.configureServiceUrl('api/prometheus/alertmanager_url', 'alertmanager');
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/services.component.ts:191]:
```suggestion
    this.configureServiceUrl('api/grafana/url', 'grafana');
    this.configureServiceUrl('ui-api/prometheus/prometheus-api-host', 'prometheus');
    this.configureServiceUrl('ui-api/prometheus/alertmanager-api-host', 'alertmanager');
```

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/services.component.ts [hunk 4]
```diff
@@ -229,6 +243,15 @@ export class ServicesComponent extends ListWithDetails implements OnChanges, OnI
...
     );
+    if (
+      this.isMgmtGateway &&
+      !this.services.find(
+        (service: CephServiceSpec) =>
+          service.service_type !== 'mgmt-gateway' && service.status.running > 0
+      )
+    ) {
+      this.isMgmtGateway = false;
+    }
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/services.component.ts:254]: doubt: why is this?

File: src/pybind/mgr/dashboard/frontend/src/app/shared/api/settings.service.ts [hunk 5]
```diff
@@ -69,7 +69,7 @@ export class SettingsService {
   }
 
   private getSettingsValue(data: any): string {
-    return data.value || data.instance || '';
+    return data.value || data.instance || data || '';
```
Comment (nizamial09) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/api/settings.service.ts:72]: ?

File: src/pybind/mgr/dashboard/controllers/prometheus.py (hunk 1 above)
Comment (Pegonzal) [reply src/pybind/mgr/dashboard/controllers/prometheus.py:145]: I thought so, but couldn't find any appropiate @nizamial09

Comment (nizamial09) [reply src/pybind/mgr/dashboard/controllers/prometheus.py:145]: i commented below the urls. will that work?

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/services.component.ts (hunk 4 above)
Comment (Pegonzal) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/services.component.ts:254]: When mgmt-gateway service is enabled and running the services urls will change therefore, I'm checking here wether the service is active or not

File: src/pybind/mgr/dashboard/controllers/prometheus.py (hunk 1 above)
Comment (Pegonzal) [reply src/pybind/mgr/dashboard/controllers/prometheus.py]: Yeah of course... Thanks Nizam

File: src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/services.component.ts (hunk 3 above)
Comment (Pegonzal) [reply src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/services/services.component.ts]: Completly missed it, thanks for pointing it out

File: src/pybind/mgr/dashboard/frontend/src/app/shared/api/settings.service.ts (hunk 5 above)
Comment (Pegonzal) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/api/settings.service.ts]: Removed it.
 It was a change to accomodate the other endpoints I added, no longer needed

**Comment by Pegonzal** [comment]
jenkins test make check

**Comment by Pegonzal** [comment]
jenkins test dashboard cephadm

**Comment by Pegonzal** [comment]
jenkins test make check

**Review by nizamial09 — APPROVED** [review]

**Review by cloudbehl — APPROVED** [review]

**Pull request merged**